text/0000-deletable-immutable-data/0000-deletable-immutable-data.md

```diff
@@ -0,0 +1,70 @@
+
+## Motivation
+`ImmutableData` is considered to be a network-owned data. As such users of SAFE Network can create it but never delete it. Moreover by their very nature `ImmutableData` are immutable and this immutablility is baked into the fact that the location where `ImmutableData` are stored (also called the `name` of `ImmutableData`) is derived from the hash of its content (also referred to as the `body`). Any change in the content will consequently produce a new name and the network will store it in a different place. Also `ImmutableData` is de-duplicated, i.e. if several users upload the same `ImmutableData`, they will all be charged but the network itself will have only one primary copy of it. However there are several cases where de-duplication is very unlikely, for instance if the content was privately encrypted or if the content was a representation of versioned directory listing of a particular user etc. In these cases when the user updates the content, they might no longer have any need for the previous content. There are also cases where one would like to delete a previously created `ImmutableData`. Currently there is neither a mechanism nor an incentive to delete unneeded `ImmutableData`. In practice we find such accumulation of never required data highly wasteful. Each churn event must account for exchange of all such data too as the network has no knowledge of what is required and what is not. Being able to delete `ImmutableData` would be nice to have and should be incentivised.
+
+## Detailed design
+When `ImmutableData` is created, the location of its storage in the network is derived from the hash of its content (`Vec<u8>`). We can imagine an approximation as:
+```rust
+// Immutable data representation:
+struct ImmutableData(Vec<u8>);
+let immut_data = ImmutableData::new(vec![255; 10]);
+
+// Name:
+let immut_data_name = XorName::new(sha512::hash(&immut_data.0).0);
+
+// Location on SAFE Network: stored by group of nodes close to immut_data_name
+
+//Conceptual storage in a node:
+let storage: HashMap<XorName, ImmutableData>;
+```
+For someone to be able to delete it there has to be some notion of ownership tied to it, otherwise `B` can delete data that was created only by `A`. We need a mechanism in place where `B` can delete data created by `A` iff `B` too had created exactly the same data. After `B` has done deleting, the data should still remain as `A` hasn't deleted it yet. Incentivising would mean that whoever deletes data gets reimbursed for playing their part in freeing up space. From this point of view, we don't want `B's` subsequent delete attempts to be legit (allowed), because `B` has already been reimbursed. When `A` finally issues a delete, a reimbursement is credited and data is finally deleted by the network.
+
+Changing what is mentioned as a conceptual storage of `ImmutableData` in a node above can help achive this:
+```rust
+enum Owner {
+    Users(Vec<sign::PublicKey>),
```

Comment:
Why not have this `Users(HashMap<sign::PublicKey, u64>) // u64 - for ref count`. Would it help with amount of data transferred curing churn?